ThingIFSDK/ThingIFSDK/TriggeredServerCodeResult.swift

```diff
@@ -35,39 +35,188 @@ public class TriggeredServerCodeResult: NSObject, NSCoding {
      - Parameter succeeded: Whether the invocation succeeded
      - Parameter returnedValue: Returned value from server code
      - Parameter executedAt: Date of the execution
-     - Parameter errorMessage: Error message of the invocation if any
+     - Parameter error: Error object of the invocation if any
      */
-    public init(succeeded: Bool, returnedValue: String?, executedAt: NSDate, errorMessage: String?) {
+    public init(succeeded: Bool, returnedValue: AnyObject?, executedAt: NSDate, error: ServerError?) {
         self.succeeded = succeeded
         self.returnedValue = returnedValue
         self.executedAt = executedAt
-        self.errorMessage = errorMessage
+        self.error = error
+    }
+    
+    public func getReturnedValue() -> AnyObject? {
+        return self.returnedValue
+    }
+    public func getReturnedValueAsString() -> String? {
+        if let str = self.returnedValue as? String {
+            return str
+        }
+        if let num = self.returnedValue as? NSNumber {
+            return String(num)
+        }
+        return nil
+    }
+    public func getReturnedValueAsBool() -> Bool? {
+        if let bool = self.returnedValue as? Bool {
```

Comment:
you don't have to check this

```swift
return self.returnedValue as? Bool
``` 
will return `nil` if returned value is not Bool